spec/Section 3 -- Type System.md

```diff
@@ -879,6 +879,8 @@ of rules must be adhered to by every Object type in a GraphQL schema.
          {"\_\_"} (two underscores).
       2. The argument must accept a type where {IsInputType(argumentType)}
          returns {true}.
+      3. If argument type is Non-Null and a default value is not defined:
+         - The `@deprecated` directive shall not be applied to this argument.
```

Comment:
Implemented here:
https://github.com/graphql/graphql-js/blob/5740cbdb58d3b50a11a29b24869a35a7ae8b6588/src/type/validate.ts#L191-L196

Reply:
and here https://github.com/graphql/graphql-js/blob/5740cbdb58d3b50a11a29b24869a35a7ae8b6588/src/type/validate.ts#L304-L308